testing/integration_testing/provisioning.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper module for dealing with mobile provisioning profiles.
+
+This module provides functionality to extract a UUID from a mobile provisioning
+file and patch it into an exports plist. The motivation for this arose when it
+was discovered that the UUID changes each time the profile is updated. Since the
+profile was pulled from a source where it was updated daily, manually specifying
+the UUID in the exports plist was not feasible.
+
+The UUID is needed to tell xcodebuild which provisioning profile to use.
+
+This should only be run on MacOS, as it uses a mac-specific
+application.
+
+"""
+
+import plistlib
+import subprocess
+
+
+def patch_provisioning_profile(plist_path, bundle_id, uuid):
```

Comment:
Might be better to have this take a named list of parameters in case we add more patches in the future?

Reply:
I'm inclined to go with YAGNI on this one, especially since there's a non-trivial detail of where in the plist to patch in the information: in this case, we're patching in a key-value pair to a root-level 'provisioningProfile' dict, but other data would likely need to go elsewhere.

Reply:
Even if we won't need other things to be patched in the provisioning profile, it's still easier to read the call if it's:
patch_provisioning_profile(plist_file, bundle_id=___, uuid=___)? There's not really a good way to differentiate it in the call otherwise. I don't feel super strongly about it so I approved regardless.

Reply:
Oh I misunderstood your suggestion. When you said named list, I thought you meant taking in a list of things to patch. 

Calling with named args does not requiring modifying the method definition, though: if a method has the signature 
def method(a, b):

Then the caller can pass in named args at their discretion. Both are valid:
method('duck', 'quack')
method(a='duck', b='quack')

Reply:
Can it require it though?

Reply:
If the signature specifies a value for something, then that will just serve as a default value. i.e. this:
def method(a=3)

Means the following are all valid:
method()  # a will be 3
method(4)
method(a=4)

There actually **is** a way to force named args in Python, though I think most people won't recognize the syntax when they see it, and it's not trivial to look up. Putting in an asterisk as an arg means the following args all have to be called as named args:

def method(a, *, b, c):

method(3, b=4, c=5)  is valid
method(3, 4, 5)  is invalid

The error also might confuse people:
"TypeError: method() takes 1 positional argument but 3 were given"

Reply:
Ah ok. Weird. I guess I'm fine to keep it as-is unless the style guide disagrees.